### PR TITLE
Likes

### PR DESCRIPTION
**Pull request by muskanbagrecha**
Please review PR for likes and watch later functionality. 

- Videos can be liked from single video page and viewed on the likes page from side navbar.
- Videos can be unliked only on the respective video page
- Video can be added to watch later playlist through the single video page and video cards as well.

PS: I am aware that I should have abstracted the util functions outside the components, and that will be addressed in the next refactoring. 

Files to check: [files](https://github.com/muskanbagrecha/video-library-react-app/pull/7/files?file-filters%5B%5D=.jsx&show-viewed-files=true)

### CODE REVIEW

File: src/Components/UI/Navigation/Navigation.jsx [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+      <form className="nav-search" onSubmit={searchHandler}>
         <input
           type="search"
           placeholder="Search for videos"
           className="input-styled"
+          onChange={(e) => {
+            setSearch(e.target.value);
+            if (e.target.value === "") {
+              filterDispatch({ type: "SEARCH", payload: "" });
+            }
```
Comment (zoef-yt) [inline src/Components/UI/Navigation/Navigation.jsx:35]: Create a separate function for this.

File: src/Components/UI/VerticalNavigation/VerticalNavigation.jsx [hunk 2]
```diff
@@ -89,20 +88,20 @@ const VerticalNavigation = ({ verticalNavOpen }) => {
...
         />
       </ul>
 
       <hr />
 
-      <ul className="menu no-list-style">{categoriesEl}</ul>
+      {/* <ul className="menu no-list-style">{categoriesEl}</ul> */}
 
-      <hr />
+      {/* <hr /> */}
```
Comment (zoef-yt) [inline src/Components/UI/VerticalNavigation/VerticalNavigation.jsx]: Remove commented code.

File: src/Context/likesContext.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+export const LikesContext = createContext([]);
+export const LikesProvider = ({ children }) => {
+  const { data: likesResponse, serverCall: likesServerCall } = useFetch();
+
+  const token = JSON.parse(localStorage.getItem("user"))?.token;
+
+  const [likedVideos, setLikedVideos] = useState([]);
+
+  useEffect(() => {
+    if (likesResponse != null && likesResponse.likes) {
```
Comment (zoef-yt) [inline src/Context/likesContext.jsx]: You can try if(likeResponse?.likes)

File: src/Context/playlistContext.jsx [hunk 4]
```diff
@@ -71,7 +71,6 @@ export const PlaylistProvider = ({ children }) => {
       url: `/api/user/playlists/${playlistId}/${videoId}`,
       headers: { authorization: token },
     });
-    console.log("deleted!");
     console.log(currPlaylistResponse);
```
Comment (zoef-yt) [inline src/Context/playlistContext.jsx:74]: Remove console.log().

File: src/Context/watchLaterContext.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+export const WatchLaterProvider = ({ children }) => {
+  const { data: watchLaterResponse, serverCall: watchLaterServerCall } =
+    useFetch();
+
+  const token = JSON.parse(localStorage.getItem("user"))?.token;
+
+  const [watchLaterVideos, setWatchLaterVideos] = useState([]);
+
+  useEffect(() => {
+    if (watchLaterResponse != null && watchLaterResponse.watchlater) {
```
Comment (zoef-yt) [inline src/Context/watchLaterContext.jsx:14]: try if (watchLaterResponse?.watchlater)

**Review by zoef-yt — APPROVED** [review]
The code is really nicely written. The UI looks amazing really love the flow of the whole app and I really like how the UI looks clean. Good job Muskan. Your good to merge.

**Review by iHarryD — APPROVED** [review]
- Loved the UI. It looks great.
- Code is nicely structured and easy to read.
- I was getting "video liked" and "video added to watch later" alerts without logging in, not sure if it was intentional.

**Comment by muskanbagrecha** [comment]
> * Loved the UI. It looks great.
> * Code is nicely structured and easy to read.
> * I was getting "video liked" and "video added to watch later" alerts without logging in, not sure if it was intentional.

Thanks for pointing it out. @iHarryD Completely missed it.

**Review by SinchanaMS — APPROVED** [review]
I really like your code, Muskaan! The names are self-explanatory and the overall code is comprehensible. Love the UI too,  it looks great! 
Suggestions have been given by the peers already. No additional ones from my end.

**Pull request merged**